include/envoy/http/codec.h

```diff
@@ -229,6 +229,12 @@ struct Http1Settings {
   bool accept_http_10_{false};
   // Set a default host if no Host: header is present for HTTP/1.0 requests.`
   std::string default_host_for_http_10_;
+  // Encode trailers in Http. By default the HTTP/1 codec drops proxied trailers.
+  // Note that this only happens when Envoy is chunk encoding which occurs when:
+  // The request is Http1.1
```

Comment:
Thanks for adding this!   Can you either link this in the API comments or reiterate the limitations there, as more people read API docs than code docs.

I'd indent these 3

//     the request is HTTP/1.1
//     the request is neither the response to a HEAD request or an HTTP upgrade
//     there is no content length in the headers


I think the content length one is a biggie.  We could optionally even call out that for folks who want consistent trailers it may be necessary to have a filter remove content length in encode/decodeHeaders

Reply:
> We could optionally even call out that for folks who want consistent trailers it may be necessary to have a filter remove content length in encode/decodeHeaders

Isn't trailers generally used only when chunk encoding though? If you already know the content length, wouldn't you just stick the information into the header? The only edge case I can imagine is that all your responses are a fixed content-length and you still need to pass in trailers based on some arbitrary data.

Reply:
Many servers will send content length with known length response bodies.  Let's say you have a set-up like the following:
[backend]  -- HTTP/2 ----  [Envoy]  ---  HTTP/1   --- client
Say your backend is sending cached content with content length headers over HTTP/2, and also sending trailers.  Your Envoy is configured to send trailers to the client over HTTP/1.1
Envoy never chunk encodes if the content length is present, so will sen the content length and strip trailers.  I think this would be very confusing to most Envoy users, who might assume if enable_trailers is true, that trailers will be propogated.

We could default strip content-length if enable-trailers is true, and there's a trailer header, but HTTP/2 doesn't require a trailer header.  We could always strip content length.  Or we could have a warning here.  I don't feel very strongly about the options, but leaving the behavior confusing where sometimes trailers are sent and sometimes not (based on the presence of the content length header) _without_ documentation seems like something we should fix.

Does the situation I describe above make sense to you?

@mattklein123 in case he has a preference

Reply:
Great call out. I would probably just go with a big warning for now (in the rendered docs) about the feature limitations. Then if someone complains we can revisit (See other docs for use of the `.. attention::` markup style).

Reply:
Fair enough, I'll update the docs. 😄